### PR TITLE
Added Documentation On Profiling Bazel Targets

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
Please review this from the perspective of someone who has never profiled anything before. 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by scveloso — DISMISSED** [review]
Looks good to me, looks straightforward.

**Review by jonathanlew — DISMISSED** [review]
LGTM!

File: docs/getting-started.md [hunk 1]
```diff
@@ -20,9 +20,8 @@
       * [Getting your Student License](#getting-your-student-license)
       * [Installing CLion](#installing-clion-1)
 * [Building and Running the Code](#building-and-running-the-code)
-   * [With CLion](#with-clion)
-   * [From the command-line](#from-the-command-line)
 * [Debugging](#debugging)
+* [Profiling](#debugging)
```
Comment (MathewMacDougall) [inline docs/getting-started.md]: Should link to profiling

File: environment_setup/setup_software.sh [hunk 2]
```diff
@@ -54,6 +54,7 @@ host_software_packages=(
     protobuf-compiler # This is required for the "NanoPb" library, which does not
                       # properly manage this as a bazel dependency, so we have 
                       # to manually install it ourselves
+    kcachegrind # This lets us view the profiles output by callgrind
```
Comment (MathewMacDougall) [inline environment_setup/setup_software.sh:57]: should we also explicitly install valgrind?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

**Review by akhilveeraghanta — DISMISSED** [review]
LG:tm:

File: environment_setup/setup_software.sh (hunk 2 above)
Comment (garethellis0) [reply environment_setup/setup_software.sh:57]: Already done: https://github.com/UBC-Thunderbots/Software/blob/master/environment_setup/setup_software.sh#L74

File: docs/getting-started.md (hunk 1 above)
Comment (garethellis0) [reply docs/getting-started.md]: Fixed.

**Review by scveloso — APPROVED** [review]

**Review by phelanm20 — APPROVED** [review]
LGTM!

File: environment_setup/setup_software.sh (hunk 2 above)
Comment (MathewMacDougall) [reply environment_setup/setup_software.sh:57]: Ah didn't see it in snap

**Review by MathewMacDougall — APPROVED** [review]
LGTM

**Pull request merged**